core/app/models/spree/order/address_book.rb

```diff
@@ -55,31 +55,19 @@ def ship_address_attributes=(attributes)
       def update_or_create_address(attributes = {})
         return if attributes.blank?
 
-        attributes = attributes.select { |_k, v| v.present? }
-
-        if user
-          address = user.addresses.build(attributes.except(:id)).check
-          return address if address.id
-        end
+        attributes.transform_values! { |v| v == '' ? nil : v }
 
         if attributes[:id]
-          address = Spree::Address.find(attributes[:id])
-          attributes.delete(:id)
+          default_address = ::Spree::Address.find(attributes[:id])
 
-          if address&.editable?
-            address.update(attributes)
-            return address
-          else
-            attributes.delete(:id)
-          end
-        end
+          if default_address&.editable?
```

Comment:
if we are expecting here `nil` value maybe we could resign from checking `if attribute[:id]` is present and search by `id` using `find_by(id: attributes[:id])` wdyt @KacperMekarski 

```suggestion
         default_address = ::Spree::Address.find_by(id: attributes[:id])

          if default_address&.editable?
            default_address.update(attributes)

            return default_address
          end
```

Reply:
Good point! It's better to avoid nested if/else conditionals

Reply:
@przemosk adjusted already.